### PR TITLE
MG-73 - Add domain ID to API endpoints

### LINKED ISSUE #73: Add domain ID to API endpoints

### PR DESCRIPTION
**Pull request by ianmuchyri**
Pull request title should be `JSSDK-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](./CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
Add's domain ID to api endpoints 
### Which issue(s) does this PR fix/relate to?
Resolves #73 
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### List any changes that modify/break current functionality
none
### Have you included tests for your changes?
yes
### Did you document any new/modified functionality?
yes
### Notes
depends on this pr from magistrala core https://github.com/absmach/magistrala/pull/2442

### CODE REVIEW

File: src/things.ts [hunk 1]
```diff
@@ -156,11 +175,49 @@ export default class Things {
...
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    try {
+      const response = await fetch(
+        new URL(
+          `${domainId}/${this.thingsEndpoint}/${thingId}/enable`,
```
[the remark this answers is not in the document]
Comment (ianmuchyri) [reply src/things.ts:199]: They are all wrapped in it:
image

File: src/things.ts [hunk 2]
```diff
@@ -95,7 +111,7 @@ export default class Things {
     }
     try {
       const response = await fetch(
-        new URL(`${this.thingsEndpoint}/bulk`, this.thingsUrl).toString(),
+        new URL(`${domainId}/${this.thingsEndpoint}/bulk`, this.thingsUrl).toString(),
```
[the remark this answers is not in the document]
Comment (ianmuchyri) [reply src/things.ts:114]: They are all wrapped in it:
image

File: src/things.ts [hunk 3]
```diff
@@ -156,11 +175,49 @@ export default class Things {
     }
   }
 
-  public async Disable (thing: Thing, token: string): Promise<Thing> {
+  public async Enable (thingId: string, domainId: string, token: string): Promise<Thing> {
+    // Enables a thing.
+    /**
+     * @method Enable - Enables a thing when provided with a valid token and thing ID.
```
Comment (Musilah) [inline src/things.ts:181]:
```suggestion
     * @method Enable - Enables a previously disabled thing when provided with a valid token, domain ID and thing ID.
```

File: src/things.ts [hunk 4]
```diff
@@ -371,32 +428,7 @@ export default class Things {
...
-        throw this.thingError.HandleError(errorRes.message, response.status)
-      }
-      const thingData: Thing = await response.json()
-      return thingData
-    } catch (error) {
-      throw error
-    }
-  }
-
-  public async IdentifyThing (thingKey: string): Promise<Thing> {
```
Comment (Musilah) [inline src/things.ts:388]: What happened to this `IdentifyThing` function. Is there a reason why we are deleting it?

Comment (ianmuchyri) [reply src/things.ts:388]: we no longer support this function on Magistrala

File: src/invitations.ts [hunk 5]
```diff
@@ -51,7 +52,7 @@ export default class Invitations {
 
     try {
       const response = await fetch(
-        new URL(this.invitationsEndpoint, this.invitationsUrl).toString(),
+        new URL(`${domainId}/${this.invitationsEndpoint}`, this.invitationsUrl).toString(),
```
Comment (Musilah) [inline src/invitations.ts:55]: When it comes to the `domainId` here why not use the `domainId` present in the invitation object? https://github.com/nyagamunene/magistrala/blob/e6f9560671db51f4449af49944a94e3ebc6df2d7/pkg/sdk/go/invitations.go#L24

File: src/invitations.ts [hunk 6]
```diff
@@ -97,11 +100,12 @@ export default class Invitations {
     }
   }
 
-  public async Invitations (queryParams: PageMetadata, token: string): Promise<InvitationsPage> {
+  public async Invitations (queryParams: PageMetadata, domainId: string, token: string): Promise<InvitationsPage> {
```
Comment (Musilah) [inline src/invitations.ts:103]: For this domainID can we also use pageMetadata for this domainID??

**Review by Musilah — APPROVED** [review]

**Review by wambui-pixel — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**